test-suite/src/test/scala/org/scalajs/testsuite/compiler/ObjectInitTest.scala

```diff
@@ -0,0 +1,40 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.compiler
+
+import org.scalajs.jasminetest.JasmineTest
+import scala.scalajs.js
+
+object ObjectInitTest extends JasmineTest {
+
+  describe("Scala Modules") {
+    when("compliant-moduleinit").
+    it("Should only execute module initializers once") {
+      val x = A.Y
+      val y = A.cs.head
+      expect(x eq null).toBeTruthy
+      expect(y eq A.Y).toBeTruthy
+      expect(Counter.c).toBe(1)
```

Comment:
Also test `y ne null`. The existing tests would succeed if everything returned `null` all the time.